C Sharp Swagger Client - tested and working

danielmarker opened the pull request:
C Sharp Client written by Mark N and I - please review

Commit:
C Sharp Swagger Client - tested and working

Commit:
renamed variables properly

tsangsimon commented:
remove CSharpSwagger.userprefs files and add that to .gitignore please,
might also want to remove everything from the /bin directory and /obj directory and again add that to be excluded in the .gitignore

Commit:
Git Ignore added

Commit:
Remove user pref file

Commit:
Removed /bin and /obj folders and excluded

Commit:
New Versioning Structure for Clients

Commit:
Update post Bug Fix for Deed Wrapper

Commit:
Post method nearly complete

Commit:
POST request works via Client

Commit:
Still awaiting Response object

Commit:
Swagger update to have String repsonse, now returning - string manipulation next to pull back the newly posted deed

Commit:
Program posts new deed then displays it using Swagger

Commit:
variable names updated

endamccormack commented on v0.1/CSharpSwagger/CSharpSwagger/Program.cs:
```diff
@@ -0,0 +1,73 @@
+
+			Console.WriteLine ("Creating Borrower");
+
+//			{
```
Are we leaving this comment in?

vaakinmade replied:
Line 21, you should be able to instantiate the DeedApplication class without explicitly declaring the "IO.Swagger.Model" since you are already using it on Line 4 ?

danielmarker replied:
@endamccormack Comment will be left so that we can prove that although extra details are there and showable in DB - an operative deed object only shows the ones required by the deed.

danielmarker replied:
@vaakinmade Fixed

endamccormack replied:
Leaving this comment here means that we are going to have to support it, i.e. if the schema changes we need to remember to come in and change this comment.

Surely it is the purpose of the swagger docs to do what you said?

vaakinmade replied:
@danielmarker Nice. similar occurrences on Lines 39 and 66 as well.

Commit:
Corrected Variable

Commit:
Client now working GET and POST from Heroku App

Commit:
Removed explicit class call for Model of DeedApplication

Kirosoft commented on v0.1/CSharpSwagger/CSharpSwagger/Program.cs:
```diff
@@ -0,0 +1,73 @@
+
+			Console.WriteLine ("New Deed created and accessible at : " + result);
+
+			string token = result.Substring (result.Length-6);
```
This would be more robust to deed-reference changes:

string token = result.Substring(result.LastIndexOf('\') + 1);

Kirosoft replied:
^^ I think there should be two slashes in the quotes above - removed by the editor

Commit:
Removed initial Hello World

endamccormack committed:
removed faulty packages refs and replace with NuGet packages

endamccormack committed:
refactored C# code for better readability and mobility
- Refactored work to suit Microsofts conventions https://msdn.microsoft.com/en-us/library/ms229043\(v\=vs.100\).aspx
- Improved logical grouping of code within methods
- Pulled out logical parts into their own method components
- Removed unnecessary additional namespaces
- Removed json payload comment

endamccormack committed:
added contents to readme for C# client

endamccormack commented:
Ended up having quite a lot of suggestions so instead of writing all the comments I made a [pull request](https://github.com/LandRegistry/dm-client-examples/pull/2), keep what you think is right or get rid of what you don't.

tsangsimon commented:
:+1:  like the code changes Enda has provided. 
Would like to have some notes on the ReadMe for Visual Studio usage as it's very specific to Xamerin right now, and most of our clients are likely to be using Visual Studio

endamccormack committed:
Added in instructions for visual studios

tsangsimon commented:
:+1:
